server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -2587,6 +2588,20 @@ private Pair<List<Long>, Integer> searchForVolumeIdsAndCount(ListVolumesCmd cmd)
         return new Pair<>(vmIds, count);
     }
 
+    private void checkEncrypted(ListVolumesCmd cmd, List<Long> ids) {
```

Comment:
It's better to add a conditional check for this. This will result in getting all the encrypted volumes from the database which can be an intensive operation.